test/unit/test_quil.py

```diff
@@ -1409,3 +1409,20 @@ def test_params_pi_and_precedence():
     prog = Program(f"RX({more_less_trivial_pi}) 0")
     exp = str(prog[0].params[0])
     assert _eval_as_np_pi(more_less_trivial_pi) == _eval_as_np_pi(exp)
+
+
+def test_copy_everything_except_instructions():
+    """Test for https://github.com/rigetti/pyquil/issues/1613"""
+    program = Program(
+        """
+DECLARE beta REAL[1]
+RZ(0.5) 0
+CPHASE(pi) 0 1
+DECLARE ro BIT[2]
+MEASURE 0 ro[0]
+MEASURE 1 ro[1]
+"""
+    )
+    program = program.copy_everything_except_instructions()
+    assert len(program.instructions) == 0  # the purpose of copy_everything_except_instructions()
+    assert len(program.declarations) == 0  # this is a view on the instructions member; must be consistent
```

Comment:
This test addresses #1611 and was added to v3 in #1614. The implementation was handled separately for v4 in #1612 and didn't include this test, which is why it's showing up now, without an implementation.